### ISSUE TITLE
Make the custom CSS validation message accessible

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#56437]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/issues/56082#issuecomment-1808295146

In the global styles > advanced panel, users can enter custom CSS. When the CSS is invalid, a vaiidation message is provided by the means of a red icon with a tooltip. Screenshot:

![Screenshot 2023-11-22 at 13 57 50](https://github.com/WordPress/gutenberg/assets/1682452/73082eaa-a330-4397-a64e-a4475b0f3a68)

This information is not accessible. Quoting from https://github.com/WordPress/gutenberg/issues/56082#issuecomment-1808295146

- No audible feedback is provided.
- The icon is just an SVG icon wrapper in a focusable `div` element.
- The focusable `div` element has no labeling and no ARIA role, it's just a silent tab stop.
- It should not be a focusable div in the first place.

I disagree with the design feedback provided in https://github.com/WordPress/gutenberg/pull/47132. Instead of adopting single-use 'ad hoc' design inspired to other tools (Codepen in this case), we should just reuste the editor components. They aim to be accessible and reusable, whilc 'ad hoc' designs defeat the purpose of usability, accessibility and maintenance. I'd argue ad-hoc implementations also defeat the purpose of reusable components in the first place. 

The error message should simply use a standard notice / warning, which has been designed to provide information in an accessible way. 

Also, whjn fixing this, some E2E test should be added to make sure the message works as intended, to avoir future regressions like https://github.com/WordPress/gutenberg/issues/56082

### Step-by-step reproduction instructions

- Go to the Site editor > Styles > More > Additional CSS
- Add this in the Additional CSS textarea: `body {background: red;}`
- Save.
- Edit the textarea again and remove the closing bracket `}`.
- Move focus away from the textarea.
- Observe a red icon appears at the bottom right of the textarea.
- Observe there is no audible feedback for screen reader users.
- Observe the red icon provides a some meaningful information only when hovered or focused.
- Observe the red icon is within a focusale `<div>` element that is not labelledd and has no ARIA role.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
